Global resource groups to avoid running heavy jobs in parallel

oldgiova opened the pull request:
Ticket: QA-983

To prevent load-biased results on tests. This way the jobs that are
    using the full load of the runner host, should go in a queue and run the
    job one at a time. This should help for reliable test results.

mzedel commented on frontend/vitest.config.ts:
```diff
@@ -66,8 +66,8 @@ export default defineConfig(() => {
+        maxWorkers: 4,
         useAtomics: true
       }
     }
```
reading your comment in the ticket I think we can replace the `pool` + `poolOptions` altogether with:
```suggestion
    maxWorkers: 4
```
or `maxWorkers: threadCount`, although locally this seems to still create more processes than the maximum 😕 (and not limit the load of the entire test suite either).

oldgiova replied:
> locally this seems to still create more processes than the maximum 😕 (and not limit the load of the entire test suite either).

Yeah, same on CI, that's why I'm experimenting...

oldgiova committed:
ci: run heavy duty jobs one at a time

To prevent load-biased results on tests. This way the jobs that are
using the full load of the runner host, should go in a queue and run the
job one at a time. This should help for reliable test results.

Ticket: QA-983

Signed-off-by: Roberto Giovanardi <roberto.giovanardi@northern.tech>

mzedel reviewed (approved):
Thanks for the investigation & let's hope this does it 🤞

lluiscampos reviewed (approved):
This is a good idea. The pipeline failed so it it was not enough.

oldgiova commented:
> This is a good idea. The pipeline failed so it it was not enough.

I hope it was because another heavy duty job was running at that time, still without the proper resource_group set. In the next days I'll add the resource_groups to all the required jobs from other projects
